tools/test_up.sh

```diff
@@ -28,31 +28,49 @@ fi
 cds="$BINDIR/crucible-downstairs"
 ct="$BINDIR/crutest"
 ch="$BINDIR/crucible-hammer"
-for bin in $cds $ct $ch; do
+dsc="$BINDIR/dsc"
+for bin in $cds $ct $ch $dsc; do
     if [[ ! -f "$bin" ]]; then
         echo "Can't find crucible binary at $bin" >&2
         exit 1
     fi
 done
 
+# Control-C to cleanup.
+trap ctrl_c INT
+function ctrl_c() {
+	echo "Stopping at your request"
+    ${dsc} cmd shutdown
+    exit 1
+}
+
+# Downstairs regions go in this directory
 testdir="/var/tmp/test_up"
 if [[ -d ${testdir} ]]; then
     rm -rf ${testdir}
 fi
 
 # Store log files we want to keep in /tmp/*.txt as this is what
 # buildomat will look for and archive
-log_prefix="/tmp/test_up"
+test_output_dir="/tmp/test_up"
+rm -rf ${test_output_dir} 2> /dev/null
+mkdir -p ${test_output_dir}
+
+log_prefix="${test_output_dir}/test_up"
 fail_log="${log_prefix}_fail.txt"
 rm -f "$fail_log"
 
 args=()
+dsc_args=()
 
 case ${1} in
     "unencrypted")
         ;;
     "encrypted")
-        args+=( --key "$(openssl rand -base64 32)" )
+        ssl_key=$(openssl rand -base64 32)
+        echo "Upstairs using SSL key: $ssl_key"
```

Comment:
nit: this isn't an SSL key, it's for AES (but more generically for on-disk encryption)

Reply:
Yeah, I guess there are other types of keys...